contribute/code.rst

```diff
@@ -0,0 +1,213 @@
+
+* In ``ubports/core/docs!1``:
+
+    ubports/core/code!2 and ubports/core/infrastructure!3 depend on this MR.
+* In ``ubports/core/code!2``:
+
+    ubports/docs!1 must be merged before this MR. ubports/core/infrastructure!3 depends on this MR.
+* In ``ubports/core/infrastructure!3``:
+
+    ubports/docs!1 and ubports/core/infrastructure!3 must be merged before this MR.
+
+Submission and review
+---------------------
+
+Now that you've checked your changeset, it's time to submit it for review! Thank you in advance for contributing to Ubuntu Touch. Here is what you can expect from us as we review your changes.
+
+We will respect you
+^^^^^^^^^^^^^^^^^^^
+
+You chose to use your time to contribute to Ubuntu Touch. That decision is never made lightly. A reviewer will treat you with respect and work with you toward your changeset becoming a part of Ubuntu Touch.
+
+Respect is a two-way street. Ubuntu Touch is a large project and there are never enough hands to do all of the work needed. It may take a while for your changeset to see any attention, and after a long wait you might come back to find a stern request for changes. Read a stern-looking message as someone trying to work as quickly as possible, not as an attempt to be rude to you. We will do the same.
+
+We will ask a lot of questions
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Reviewers gain a better understanding of the code you're changing by asking you questions about how it works. They may know what your code does already, but they'll still ask. If you can explain how your code works, it's more likely that you have done the proper testing to ensure it works. If you are not able to articulate why something works, it is a red flag. It is very likely that the change is working around a different bug that will come back to haunt us in the future. Our time would be better spent fixing the real bug than trying to make workarounds.
+
+We will ask for changes
+^^^^^^^^^^^^^^^^^^^^^^^
+
+Another person looking at your code will likely see problems with it that you have missed. Whether those problems are inefficiencies, style issues, or new bugs, they are more likely to be found by your reviewers than by you. Don't worry if you get a book of requested changes back from your reviewer. Nobody is perfect. Finding and fixing these potential issues leads to a faster and more stable Ubuntu Touch, it is not an insult to your skills as a developer.
+
+We may ask for huge changes
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This is distinct from the previous section. Sometimes your reviewers will look at your change and realize that a much deeper issue exists in the component. Sometimes you start out developing a feature with the wrong assumptions in mind and end up with an unwieldy mess of spaghetti code. Either way: you took the wrong path to your destination. Whatever the reasons may be, your reviewer will politely point out the potential problems with your change and constructively suggest a new direction for you to take.
```

Comment:
maybe the following could take out some pressure, by making it less personal: "you took the wrong path to your destination" -> "this is the wrong path to the destination"